### PR TITLE
Check if String/Optional String is empty

### PR DESCRIPTION
**Pull request by aashishdhawan**
1. Check is optional String `isNilOrEmpty`.
2. Checks if string `isEmptyOrWhiteSpace`

### CODE REVIEW

**Comment by zenangst** [comment]
@aashishdhawan you need to fix the conflicts again :sun_with_face:

**Comment by aashishdhawan** [comment]
@zenangst  Done.

File: Source/Shared/Extensions/String+Empty.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
+//
+//  String+Empty.swift
+//  Sugar
+//
+//  Created by Aashish Dhawan on 18/03/16.
+//  Copyright © 2016 Hyper Interaktiv AS. All rights reserved.
+//
```
Comment (zenangst) [inline Source/Shared/Extensions/String+Empty.swift]: Could you remove this like you did for the other PRs?

Comment (aashishdhawan) [inline Source/Shared/Extensions/String+Empty.swift]: Sure. My bad. this slipped in.

**Comment by aashishdhawan** [comment]
@zenangst Closing this. Let me do this again Please

**Pull request closed without merging**